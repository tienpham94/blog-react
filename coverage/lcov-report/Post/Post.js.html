<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Post/Post.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Post</a> Post.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Link } from 'react-router-dom'
&nbsp;
import styles from './Post.css'
&nbsp;
const Post = (props) =&gt; (
  &lt;div&gt;
    &lt;h2 className={styles.title}&gt;
      {props.titleLink
      ? &lt;Link className={styles.link} to={`/post/${props.slug}`}&gt;{props.title}&lt;/Link&gt;
      : props.title}
    &lt;/h2&gt;
    &lt;p className={styles.content}&gt;{props.excerpt}&lt;/p&gt;
  &lt;/div&gt;
)
&nbsp;
export default Post
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Mar 29 2018 11:50:56 GMT+0300 (EEST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
