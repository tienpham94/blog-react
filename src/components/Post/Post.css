.title {
  font-size: 3rem;
  color: #2F89DE;
}

.link {
  color: #2F89DE;
}

.link:focus,
.link:hover {
  color: #30699E;
  transition: color 0.3s ease;
}

.content {
  font-size: 1.8rem;
  line-height: 1.7;
}
