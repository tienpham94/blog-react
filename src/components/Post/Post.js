import React from 'react'
import { Link } from 'react-router'

import styles from './Post.css'

const Post = (props) => (
  <div>
    <h2 className={styles.title}>
      {props.titleLink
      ? <Link className={styles.link} to={`/post/${props.slug}`}>{props.title}</Link>
      : props.title}
    </h2>
    <p className={styles.content}>{props.excerpt}</p>
  </div>
)

export default Post
