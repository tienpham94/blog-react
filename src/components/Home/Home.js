import React from 'react'

import Post from '../Post/Post'

const Home = ({posts}) => (
  <div>
    {posts.length
      ? posts.map(post => <Post {...post} key={post.slug} titleLink />) : <p>No posts here</p>
    }
  </div>
)

export default Home
