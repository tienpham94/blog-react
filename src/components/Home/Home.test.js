import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'
import Post from '../Post/Post'


test('component', () => {
  const posts = {
    posts: [{
      title: 'How to build a progressive web app with React',
      slug: 'how-to-build-a-progressive-web-app-with-react',
      excerpt: 'One of the easiest ways to get a job',
      content: 'One of the easiest ways to get a job in the early GitHub days...'
    }]
  };
  const component = shallow(<Home posts={posts.posts} />)
  const tree = toJson(component)
  expect(component.find(Post).length).toBe(1)
  expect(tree).toMatchSnapshot()
})

test('cpme', () => {
  const component = shallow(<Home posts={[]} />)
  const tree = toJson(component)

  expect(component.find(Post).length).toBe(0)
  expect(component.contains('No posts here')).toBe(true)
  expect(tree).toMatchSnapshot()
})
