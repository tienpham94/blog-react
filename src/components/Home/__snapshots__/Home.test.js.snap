exports[`test component 1`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days..."
    excerpt="One of the easiest ways to get a job"
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true} />
</div>
`;

exports[`test cpme 1`] = `
<div>
  <p>
    No posts here
  </p>
</div>
`;
